libhb/qsv_common.c

```diff
@@ -1542,6 +1542,15 @@ int hb_qsv_profile_parse(hb_qsv_param_t *param, hb_qsv_info_t *info, const char
 
             case MFX_CODEC_HEVC:
                 profile = hb_triplet4key(hb_qsv_h265_profiles, profile_key);
+
+                /* HEVC10 supported starting from KBL/G6 */
+                if (profile->value == MFX_PROFILE_HEVC_MAIN10 &&
+                    qsv_hardware_generation(hb_get_cpu_platform()) < QSV_G6)
+                {
+                    hb_log("HEVC Main10 is not supported on this platform");
```

Comment:
I wonder if we also need to update hb_video_encoder_get_profiles. The WinGUI uses this for getting profiles. (oddly I don't see QSV in there so I wonder how that even works)

Point being, we could make sure main10 is not available in the gui on pre kaby hardware.

Thoughts?